vsphere-mixin/panels.libsonnet

```diff
@@ -0,0 +1,1428 @@
+        commonlib.panels.generic.stat.info.new(
+          'Clustered VMs off',
+          targets=[t.clusteredVMsOffCount],
+          description='The number of virtual machines currently powered off that belong to a cluster in the datacenter.'
+        )
+        + stat.options.withGraphMode('none'),
+
+      clusteredVMsSuspendedStatus:
+        commonlib.panels.generic.stat.info.new(
+          'Clustered VMs suspended',
+          targets=[t.clusteredVMsSuspendedCount],
+          description='The number of virtual machines currently in a suspended state that belong to a cluster in the datacenter.'
+        )
+        + stat.options.withGraphMode('none'),
+
+      clusteredVMTemplatesCountStatus:
+        commonlib.panels.generic.stat.info.new(
+          'Clustered VM templates',
+          targets=[t.clusteredVMTemplatesCount],
+          description='The number of virtual machine templates that belong to a cluster in the datacenter.'
+        )
+        + stat.options.withGraphMode('none'),
+
+      clusteredHostsActiveStatus:
+        commonlib.panels.generic.stat.info.new(
+          'Clustered active ESXi hosts',
+          targets=[t.clusteredHostsActiveCount],
+          description='The number of ESXi hosts that are currently running (responding and not in maintenance mode) that belong to a cluster within the datacenter.'
+        )
+        + stat.options.withGraphMode('none'),
+
+      clusteredHostsInactiveStatus:
+        commonlib.panels.generic.stat.info.new(
+          'Clustered inactive ESXi hosts',
+          targets=[t.clusteredHostsInactiveCount],
+          description='The number of ESXi hosts that are currently not running (not responding or in maintenance mode) that belong to a cluster within the datacenter.'
+        )
+        + stat.options.withGraphMode('none'),
+
+      topCPUUtilizationClusters:
+        commonlib.panels.memory.timeSeries.usagePercent.new(
+          'Top CPU utilization by cluster',
+          targets=[t.topCPUUtilizationClusters],
+          description='The clusters with the highest CPU utilization percentage in the datacenter.'
+        )
+        + g.panel.timeSeries.options.legend.withDisplayMode('table')
+        + g.panel.timeSeries.standardOptions.withUnit('percent'),
+
+      topMemoryUtilizationClusters:
+        commonlib.panels.memory.timeSeries.usagePercent.new(
+          'Top memory utilization by cluster',
+          targets=[t.topMemoryUtilizationClusters],
+          description='The clusters with the highest memory utilization percentage in the datacenter.'
+        )
+        + g.panel.timeSeries.options.legend.withDisplayMode('table')
+        + g.panel.timeSeries.standardOptions.withUnit('percent'),
+
+      clustersTable:
+        commonlib.panels.generic.table.base.new(
+          'Clusters table',
+          targets=[
+            t.effectiveCPUClusters + g.query.prometheus.withFormat('table')
+            + g.query.prometheus.withRange(true)
```

Comment:
nit: you could use withRefId() here to name each query with meaningful name instead of Value #A-Z. Not required, but easy to follow in latter transforms what is being changed.

Reply:
I changed all of these as well even though I probably didn't need to 😄